tests/_support/_bootstrap.php

```diff
@@ -7,8 +7,8 @@
 $_SERVER['CI_ENVIRONMENT'] = 'testing';
 define('ENVIRONMENT', 'testing');
 
-// Path to the front controller (this file)
-define('FCPATH', getcwd().'/public'.DIRECTORY_SEPARATOR);
+// path to the directory that holds the front controller (index.php)
+define('FCPATH', realpath(__DIR__.'/../../') . '/public'.DIRECTORY_SEPARATOR);
```

Comment:
As long as we're making changes here, we should honor the `public` folder's name that's found in `application/Config/Paths.php`, `$publicDirectory`.

Reply:
TL:DR:

- loading the application/Config/Paths.php in this case is not trivial because it has a dependency on FCPATH already being set.
- I would recommend leaving it the string `'public'`. That way if anyone does change it in Config/Paths they will get "file not found" in all their tests.
- `'public'` was a string in the first place, the issues below are probably why
 
Details:

I looked into this and in order to load the `Config\Paths()` the `public/index.php` has to do this:

`define('FCPATH', __DIR__.DIRECTORY_SEPARATOR);
$pathsPath = FCPATH . '../application/Config/Paths.php';
require $pathsPath;
$paths = new Config\Paths()
// then $paths->publicDirectory when you want to see 'public'`

in the test `_bootstrap.php` we are trying to _create_ FCPATH. 

Unfortunately, without FCPATH created no tests will pass (tried this too).

Reply:
That's fair. Ignore me on this one. :)